### PR TITLE
Remove extra dereference from the borrow_deref_ref lint example

### PR DESCRIPTION
**Pull request by stanislav-tkach**
I don't think such minor change should be mentioned in the changelog.

changelog: none

### CODE REVIEW

**Comment by lengyijun** [comment]
I prefer `&**` than `&*` .

**Comment by stanislav-tkach** [comment]
@lengyijun This is from the lint description:
```rust
...
fn foo(_x: &str) {}

// Before:
let a: &String = &* s;
foo(&*s); // 1

// Use instead:
let a: &String = s;
foo(&**s); // 2
```
In both cases we have the same type, but in the first (problematic) case the `&*s` construction is used and in the second (fixed) one it is different for some reason. It is confusing: I spent several minutes figuring out why it is different and as we can see, I came to a wrong conclusion. I don't mind using `&**`, but I propose to use the same construction in both places, because it is unrelated to the lint.

File: clippy_lints/src/borrow_deref_ref.rs [hunk 1]
```diff
@@ -42,7 +42,7 @@ declare_clippy_lint! {
     /// # fn foo(_x: &str) {}
     /// # let s = &String::new();
     /// let a: &String = s;
-    /// foo(&**s);
+    /// foo(&*s);
```
Comment (giraffate) [inline clippy_lints/src/borrow_deref_ref.rs:45]: It would be better to remove the `foo` function itself.
```suggestion
```

**Comment by giraffate** [comment]
Okay. There are two warnings in the example, first is `borrow_deref_ref` and second is `explicit_auto_deref`. So it would be better to remove second warning.

File: clippy_lints/src/borrow_deref_ref.rs (hunk 1 above)
Comment (stanislav-tkach) [reply clippy_lints/src/borrow_deref_ref.rs]: Makes sense, done.

**Comment by giraffate** [comment]
@bors r+

Thanks!

**Pull request merged**